draft-ietf-httpbis-client-hints.md

```diff
@@ -169,6 +200,17 @@ Implementers ought to consider both user and server controlled mechanisms and po
 Implementers SHOULD support Client Hints opt-in mechanisms and MUST clear persisted opt-in preferences when any one of site data, browsing history, browsing cache, or similar, are cleared.
 
 
+## Deployment and Security Risks
+Deployment of new request headers requires several considerations:
+
+  - Potential conflicts due to existing use of field name
+  - Properties of the data communicated in field value
+
+Specifications and features relying on Client Hints are advised to prefix request header names with a "Sec-" prefix. User agents, such as web browsers, ensure such prefixed headers can only be emitted by the browser, and application content, such as scripts, are forbidden from emitting such headers {{FETCH}}. This provides assurance that, when transmitted by user agents, the type, format and value of data communicated in such request headers are enforced by the user agent.
```

Comment:
Once again -- this text assumes that EVERY Client Hint is designed such that making it settable from code is an attack, and that there isn't any value in allowing it happen. I'd be much more comfortable saying that specs defining (not relying upon, btw) a Client Hint _carefully consider_ whether it should be exposed to code, and if it's a risk, to prefix with `Sec-`.

Reply:
changed to "Specifications and features ... are advised to carefully consider prefixing..."